src/components/treeActivity/index.tsx

```diff
@@ -35,19 +35,146 @@ const EntryMessage = styled(Paragraph)`
+const StewardshipActivityDropdownContainer = styled.div`
+  margin: -10px 0 -5px 5px;
+`;
+
+const StewardshipActivityDropdown = styled.div`
+  font-size: 15px;
+  color: ${MID_GREEN};
+`;
+
+const CenteredPagination = styled(Pagination)`
+  display: flex;
+  justify-content: center;
+  & .ant-pagination-item-active {
+    background-color: white;
+  }
+  & .ant-pagination-item-active :hover {
+    background-color: ${DARK_GREEN};
+  }
+`;
+
 interface TreeActivityProps {
   readonly stewardship: TreeCare[];
-  readonly limit?: number;
 }
 
-const TreeActivity: React.FC<TreeActivityProps> = ({ stewardship, limit }) => {
+const TreeActivity: React.FC<TreeActivityProps> = ({ stewardship }) => {
+  const months = [
+    'January',
+    'February',
+    'March',
+    'April',
+    'May',
+    'June',
+    'July',
+    'August',
+    'September',
+    'October',
+    'November',
+    'December',
+  ];
+
+  // convert a month from its three letter abbreviation (e.g. 'Jan') to its long form (e.g. 'January')
+  const monthShortToLong = (shortMonth: string): string => {
+    for (const month of months) {
+      if (shortMonth.substring(0, 3) === month.substring(0, 3)) {
+        return month;
+      }
+    }
+    return '';
+  };
+
+  // sort two dates by a combination of year and month
+  const sortByYearMonth = (
+    yearMonth1: { label: string },
+    yearMonth2: { label: string },
+  ) => {
+    const label1 = yearMonth1.label;
+    const label2 = yearMonth2.label;
+
+    const year1 = label1.slice(-4);
+    const month1 = months.indexOf(label1.split(' ')[0]);
+    const year2 = label2.slice(-4);
+    const month2 = months.indexOf(label2.split(' ')[0]);
+
+    if (year1 === year2 && month1 === month2) {
+      return 0;
+    } else if (year1 < year2 || (year1 === year2 && month1 < month2)) {
+      return -1;
+    } else {
+      return 1;
+    }
+  };
+
+  const [selectedMonthYear, setSelectedMonthYear] = useState(
+    new Date().toLocaleString('default', {
+      month: 'short',
+      year: 'numeric',
+    }),
+  );
+  const selectedMonthYearStewardship: TreeCare[] = stewardship.filter(
+    (entry) =>
+      // choose stewardship activities with the selected month and year
+      entry.date.substring(0, 3) === selectedMonthYear.substring(0, 3) &&
+      entry.year === parseInt(selectedMonthYear.slice(-4), 10),
+  );
+  const monthYearOptions = stewardship
+    // remove activities with duplicate year and date
+    .filter(
+      (entry, index, self) =>
+        index ===
+        self.findIndex(
+          (e) =>
+            e.year === entry.year &&
+            e.date.substring(0, 3) === entry.date.substring(0, 3),
+        ),
+    )
+    // turn the filtered array into an array of labels and values
+    .map((entry) => {
+      return {
+        label: monthShortToLong(entry.date) + ' ' + entry.year,
+        value: entry.date.substring(0, 3) + ' ' + entry.year,
+      };
+    });
+  // append the current month and year to the options list
```

Comment:
❓ What happens if one of the stewardship activities took place in the current month and year?

Reply:
Oh interesting. I think I fixed this by flipping the orders in which the current month/year was added to `monthYearOptions` and the duplicate elements in `monthYearOptions` were removed.